tests/test_spstats.py

```diff
@@ -0,0 +1,49 @@
+"""
+Functions to test the spatial statistics.
+
+"""
+from __future__ import annotations
+import numpy as np
+import xdem
+from xdem import examples
+import geoutils as gu
+
+
+def load_diff() -> tuple[gu.georaster.Raster, np.ndarray] :
+    """Load example files to try coregistration methods with."""
+    examples.download_longyearbyen_examples(overwrite=False)
+
+    reference_raster = gu.georaster.Raster(examples.FILEPATHS["longyearbyen_ref_dem"])
+    to_be_aligned_raster = gu.georaster.Raster(examples.FILEPATHS["longyearbyen_tba_dem"])
+    glacier_mask = gu.geovector.Vector(examples.FILEPATHS["longyearbyen_glacier_outlines"])
+
+    metadata = {}
+    aligned_raster, _ = xdem.coreg.coregister(reference_raster, to_be_aligned_raster, method="amaury", mask=glacier_mask,
+                                     metadata=metadata)
+
+    diff = xdem.spatial_tools.subtract_rasters(reference_raster,aligned_raster)
```

Comment:
This is the most correct way to do it, but for speed, I suggest adding `resampling_method="nearest"` as a keyword argument. These specific DEMs are already in the same grid, so it will speed up the test slightly!

Reply:
If the two DEMs are on the same grid, there should be no interpolation then and the type of interpolation should not matter. Is there a check for this in substract_rasters?

Reply:
@adehecq that is not checked but maybe it should be. I just call `Raster.reproject(...., resampling=resampling_method)`.

I experimented a bit with geoutils:
```python

import time

import geoutils as gu
import rasterio as rio

# I use the KH-9 DEM you sent me which is quite big
dem = gu.georaster.Raster("/home/erik/Downloads/DZB1208-500032L004006/DZB1208-500032L004006-DEM.tif")

# I reproject the DEM to its own exact bounds using cubic spline
# This is a proxy for if a DEM with the exact same bounds as a reference DEM should (not) be reprojected.
time_before = time.time()
dem2 = dem.reproject(dem, resampling=rio.warp.Resampling.cubic_spline)
time_after = time.time()
print(f"{time_after - time_before:.2f} s")

```
Which returned: 
```
4.98 s
```
I would argue that this is a limitation in either `georaster` or `rasterio`. Why would resampling be done if the source and destination bounds/crs/resolution is the same??

I see two approaches to circumvent this: either rewrite the `subtract_rasters()` function a little, post an associated issue on `geoutils`, or both. What do you think?

Reply:
> > Nice!
> > Another suggestion: It would be really nice to have an example in the README for variogram calculations with default settings. Could this be a worthy addition?
> 
> Agreed! There is a lot of things in this PR so it's difficult to check it all, but it would be nice to be able to test the different features. Anyway, at some point we should definitely write some documentation/how-to.

Yes, this will come when all is done! For now, trying to make everything consistent, working and finish the related tests :)

Reply:
> @adehecq that is not checked but maybe it should be. I just call `Raster.reproject(...., resampling=resampling_method)`.
> 
> I experimented a bit with geoutils:
> 
> ```python
> import time
> 
> import geoutils as gu
> import rasterio as rio
> 
> # I use the KH-9 DEM you sent me which is quite big
> dem = gu.georaster.Raster("/home/erik/Downloads/DZB1208-500032L004006/DZB1208-500032L004006-DEM.tif")
> 
> # I reproject the DEM to its own exact bounds using cubic spline
> # This is a proxy for if a DEM with the exact same bounds as a reference DEM should (not) be reprojected.
> time_before = time.time()
> dem2 = dem.reproject(dem, resampling=rio.warp.Resampling.cubic_spline)
> time_after = time.time()
> print(f"{time_after - time_before:.2f} s")
> ```
> 
> Which returned:
> 
> ```
> 4.98 s
> ```
> 
> I would argue that this is a limitation in either `georaster` or `rasterio`. Why would resampling be done if the source and destination bounds/crs/resolution is the same??
> 
> I see two approaches to circumvent this: either rewrite the `subtract_rasters()` function a little, post an associated issue on `geoutils`, or both. What do you think?

I think this should be raised as an issue in geoutils

Reply:
@rhugonnet [I added an issue in geoutils](https://github.com/GlacioHack/GeoUtils/issues/102)!

Reply:
Great, that's what I was going to suggest!